app/controllers/display_ads_controller.rb

```diff
@@ -0,0 +1,30 @@
+class DisplayAdsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[for_display], unless: -> { current_user }
+  CACHE_EXPIRY_FOR_DISPLAY_ADS = 15.minutes.to_i.freeze
+
+  def for_display
+    skip_authorization
+    set_cache_control_headers(CACHE_EXPIRY_FOR_DISPLAY_ADS) unless session_current_user_id
+
+    if params[:placement_area]
+
+      if params[:article_id].present?
+        @article = Article.find(params[:article_id])
+      end
+
+      @display_ad = DisplayAd.for_display(
+        area: params[:placement_area],
+        user_signed_in: user_signed_in?,
+        organization_id: @article&.organization_id,
+        permit_adjacent_sponsors: @article&.decorate&.permit_adjacent_sponsors? || true,
```

Comment:
For what it's worth, this `||` means `permit_adjacent_sponsors` can _never_ be `false`, which probably isn't intended.

Something like `ArticleDecorator.new(nil).permit_adjacent_sponsors? || true` can be made to work, but would require some modifications in the decorator, as it currently assumes `user_id` is _always_ available. (And, if we're already there, `|| true` can be handled there as well.)

```diff
  def permit_adjacent_sponsors?
+  return true unless respond_to?(:user_id) && user_id.present?
    author_ids = [user_id] + co_author_ids
    Users::Setting.where(user_id: author_ids).all?(&:permit_adjacent_sponsors)
  end
```

Reply:
Whoops 🤦🏽‍♀️ I'll look into this.